ThePortugueseMan.HabitTracker/Habit_DataBase/DataBaseCommands.cs

```diff
@@ -0,0 +1,394 @@
+﻿namespace DataBaseLibrary;
+using Microsoft.Data.Sqlite;
+using System;
+using System.Globalization;
+
+
+public class DataBaseCommands
+{
+    static string connectionString = @"Data Source=habit-Tracker.db";
+    static string s_MainTableName = "HabitsTable";
+
+    //if the main table doesn't exist, it's created
+    public void Initialization(string? mainTableName)
+    {
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+
+            tableCmd.CommandText =
+                @$"CREATE TABLE IF NOT EXISTS {mainTableName}" +
+                    "(Id INTEGER PRIMARY KEY AUTOINCREMENT," +
+                    "HabitTableName TEXT," +
+                    "HabitUnit TEXT)";
+
+            tableCmd.ExecuteNonQuery();
+
+            connection.Close();
+        }
+    }
+    //creates a new subtable represeting an habit - each entry has a string date and an int quantity
+    public void CreateSubTable(string? tableName)
+    {
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+
+            tableCmd.CommandText =
+                @$"CREATE TABLE IF NOT EXISTS {tableName}" +
+                    "(Id INTEGER PRIMARY KEY AUTOINCREMENT," +
+                    "Date TEXT," +
+                    "Quantity INTEGER)";
+
+            tableCmd.ExecuteNonQuery();
+
+            connection.Close();
+        }
+    }
+
+    //Insert log to subtable - overload based on data types
+    public void Insert(string? subTableName, string? date, int quantity)
+    {
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+
+            tableCmd.CommandText =
+                $"INSERT INTO {subTableName}(date, quantity) VALUES ('{date}',{quantity})";
+
+            tableCmd.ExecuteNonQuery();
+
+            connection.Close();
+        }
+    }
+    //Insert habit to main table - overload based on data types
+    public void Insert(string? mainTableName,string? subTableName, string? habitUnit)
+    {
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+
+            tableCmd.CommandText =
+                $"INSERT INTO {mainTableName}(HabitTableName, HabitUnit)" +
+                $" VALUES ('{subTableName}','{habitUnit}')";
+
+            tableCmd.ExecuteNonQuery();
+
+            connection.Close();
+        }
+    }
+    //checks if there is an entry at Id = index
```

Comment:
Please check the necessity of this comment. 😊

Reply:
Removed!